internal/controllers/inventory_controller.go

```diff
@@ -1439,16 +1465,55 @@ func (t *reconcilerTask) deployServer(ctx context.Context, serverName string) (u
 		}...)
 	}
 
-	// Common evn for server deployments
-	envVars = append(envVars,
-		corev1.EnvVar{
+	proxyArgs := []string{
+		fmt.Sprintf("--secure-listen-address=0.0.0.0:%d", utils.DefaultProxyPort),
+		fmt.Sprintf("--upstream=http://127.0.0.1:%d/", utils.DefaultContainerPort),
+		"--logtostderr=true",
+		"--tls-cert-file=/secrets/tls/tls.crt",
+		"--tls-private-key-file=/secrets/tls/tls.key",
+		"--tls-min-version=VersionTLS12",
+		"--v=10",
+	}
+
+	if utils.IsOAuthEnabled(t.object) {
```

Comment:
Could we move this in its own function as well? This `deployServer` is becoming quite large.

Reply:
fixed